File: SoftwaretechnikProjekt/JavaJsonMQTest/src/entities/Schnorchel.java
package entities;


import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Schnorchel {
    private int alter;
    private String name;
    private List<Hobby> hobbies = new ArrayList<Hobby>();
    private Map<String, Boolean> faehigkeiten = new HashMap<String,Boolean>();

    public int getAlter() {
		return alter;
	}

	public void setAlter(int alter) {
		this.alter = alter;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		this.name = name;
	}

	public List<Hobby> getHobbies() {
		return hobbies;
	}

	public void setHobbies(List<Hobby> hobbies) {
		this.hobbies = hobbies;
	}

	public Map<String, Boolean> getFaehigkeiten() {
		return faehigkeiten;
	}

	public void setFaehigkeiten(Map<String, Boolean> faehigkeiten) {
		this.faehigkeiten = faehigkeiten;
	}

	@Override
    public String toString()
    {
    	String hob = "";
    	for (Hobby h: hobbies) {
    		hob += " " + h.hobbyname;
    	}
    	String fk = "";
    	for (String f: faehigkeiten.keySet()) {
    		fk += " "+f;
    	}
    	
    	Hobby[] h = (Hobby[]) hobbies.toArray(new Hobby[hobbies.size()]);
        return String.format("Ich bin ein %d Jahre alter Schnorchel und heisse %s. Meine Hobbies sind %s, ich kann %s", 
            alter, name, hob, fk);
    }

}

File: SoftwaretechnikProjekt/Client/CommunicatioClient/ObjectTest/Schnorchel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ObjectTest
{
    class Schnorchel
    {
        public int alter { get; set; }
        public string name { get; set; }
        public List<Hobby> hobbies = new List<Hobby>();
        public Dictionary<string, bool> faehigkeiten = new Dictionary<string,bool>();

        public override string ToString()
        {
            return string.Format("Ich bin ein {0} Jahre alter Schnorchel und heisse {1}. Meine Hobbies sind {2}, ich kann {3}", 
                alter, name, string.Join(", ",hobbies.Select(p => p.hobbyname)),
                string.Join("; ",faehigkeiten.ToList().Select(x => x.ToString()))
                );
        }
    }
}

